### ISSUE TITLE
Detectnet Spacenet: Model fails when resizing image from original kitti size to 1280x1280

### ISSUE THREAD

**Issue by 00krishna** [NVIDIA/DIGITS#2041]
Hey Folks.
So I imported the spacenet data as a custom dataset, as per the instructions. I also made sure to reference the custom labels as `dontcare,building` when the dataset was imported. So that worked fine. However, when adjusting the Kitti implementation of Detectnet to the Spacenet data, I ran into a strange `ValueError` upon changing the image sizes in the model `prototxt`.

As per the blog post I resized the images to 1280x1280 while importing them to the LMDB, no padding. There are about 6 locations where the image size is mentioned (3 at the top of the file and 3 at the bottom). I tried to change the size in 2 different ways, but neither succeeded. The first one trained a model but I got 0 mAP. In the second one the model just crashed. 

I read through any number of issues and this very useful discussion https://www.bountysource.com/issues/36984553-how-can-i-use-detectnet-for-custom-size-data . Seems like all that should be needed is a change in the image sizes and it should work, but for some reason it is not that easy. 

So for the first attempt I changed only the image specs at the top, so that means i changed lines 57, 58, 79, 80, 118, and 119. That was all that I changed. NOTE: I APPLIED THE RANDOM CROPPING OF TRAINING DATA ON LINES 79, 80. This model trained, but it did not get any detections even after 50 epochs. The specific file changes were:

```
layer {
  name: "deploy_data"
  type: "Input"
  top: "data"
  input_param {
    shape {
      dim: 1
      dim: 3
      dim: 1280
      dim: 1280
    }
  }
  include: { phase: TEST not_stage: "val" }
}
# Data transformation layers
layer {
  name: "train_transform"
  type: "DetectNetTransformation"
  bottom: "data"
  bottom: "label"
  top: "transformed_data"
  top: "transformed_label"
  detectnet_groundtruth_param: {
    stride: 16
    scale_cvg: 0.4
    gridbox_type: GRIDBOX_MIN
    coverage_type: RECTANGULAR
    min_cvg_len: 20
    obj_norm: true
    image_size_x: 512
    image_size_y: 512
    crop_bboxes: true
    object_class: { src: 1 dst: 0} # obj class 1 -> cvg index 0
  }
   detectnet_augmentation_param: {
    crop_prob: 1
    shift_x: 32
    shift_y: 32
    flip_prob: 0.5
    rotation_prob: 0
    max_rotate_degree: 5
    scale_prob: 0.4
    scale_min: 0.8
    scale_max: 1.2
    hue_rotation_prob: 0.8
    hue_rotation: 30
    desaturation_prob: 0.8
    desaturation_max: 0.8
  }
  transform_param: {
    mean_value: 127
  }
  include: { phase: TRAIN }
}
layer {
  name: "val_transform"
  type: "DetectNetTransformation"
  bottom: "data"
  bottom: "label"
  top: "transformed_data"
  top: "transformed_label"
  detectnet_groundtruth_param: {
    stride: 16
    scale_cvg: 0.4
    gridbox_type: GRIDBOX_MIN
    coverage_type: RECTANGULAR
    min_cvg_len: 20
    obj_norm: true
    image_size_x: 1280
    image_size_y: 1280
    crop_bboxes: false
    object_class: { src: 1 dst: 0} # obj class 1 -> cvg index 0
  }
  transform_param: {
    mean_value: 127
  }
  include: { phase: TEST stage: "val" }
}
```

Next I tried the same prototext model, but this time I also change the image sizes at the bottom of the file. So that means lines 2504, 2519, and 2545--as pictured below. 

```
# Cluster bboxes
layer {
    type: 'Python'
    name: 'cluster'
    bottom: 'coverage'
    bottom: 'bboxes'
    top: 'bbox-list'
    python_param {
        module: 'caffe.layers.detectnet.clustering'
        layer: 'ClusterDetections'
        param_str : '1280, 1280, 16, 0.6, 3, 0.02, 22, 1'
    }
    include: { phase: TEST }
}

# Calculate mean average precision
layer {
  type: 'Python'
  name: 'cluster_gt'
  bottom: 'coverage-label'
  bottom: 'bbox-label'
  top: 'bbox-list-label'
  python_param {
      module: 'caffe.layers.detectnet.clustering'
      layer: 'ClusterGroundtruth'
      param_str : '1280, 1280, 16, 1'
  }
layer {
    type: 'Python'
    name: 'mAP'
    bottom: 'bbox-list-scored'
    top: 'mAP'
    top: 'precision'
    top: 'recall'
    python_param {
        module: 'caffe.layers.detectnet.mean_ap'
        layer: 'mAP'
        param_str : '1280, 1280, 16'
    }
    include: { phase: TEST stage: "val" }
```

When I made this change though, the model crashed with an error in the caffe log file:
```
Traceback (most recent call last):
File "/usr/local/python/caffe/layers/detectnet/clustering.py", line 66, in forward
bbox = cluster(self, data0, bottom[1].data)
File "/usr/local/python/caffe/layers/detectnet/clustering.py", line 232, in cluster
boxes[i, 0:r, 0:c] = boxes_cur_image
ValueError: could not broadcast input array from shape (52,4) into shape (50,4)
```

Does anyone know where the error is? I have been struggling with this for like 3 days now, so hopefully someone can provide some insight into where the error is or what is the right way to set these image sizes? Any help would really be appreciated @jbarker-nvidia @gheinrich @lukeyeager .

**Comment by 00krishna** [comment]
Looks like this might be related to NVIDIA/caffe#228, NVIDIA/caffe#238. I will try that answer.

**Comment by 00krishna** [comment]
Yes, seems like this was related to changing the MAX_BOXES parameter. I solved this using the steps in #2042 .
